presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -143,111 +127,71 @@ private boolean advanceToNextRowGroup()
 
         nextRowInGroup = 0L;
         currentGroupRowCount = currentBlockMetadata.getRowCount();
-        columnReadersMap.clear();
         initializeColumnReaders();
         return true;
     }
 
-    public Block readArray(Type type, List<String> path)
-            throws IOException
-    {
-        return readArray(type, path, new IntArrayList());
-    }
-
-    private Block readArray(Type type, List<String> path, IntList elementOffsets)
+    private ColumnChunk readArray(GroupField field)
             throws IOException
     {
-        List<Type> parameters = type.getTypeParameters();
+        List<Type> parameters = field.getType().getTypeParameters();
         checkArgument(parameters.size() == 1, "Arrays must have a single type parameter, found %d", parameters.size());
-        path.add(ARRAY_TYPE_NAME);
-        Type elementType = parameters.get(0);
-        Block block = readBlock(ARRAY_ELEMENT_NAME, elementType, path, elementOffsets);
-        path.remove(ARRAY_TYPE_NAME);
-
-        if (elementOffsets.isEmpty()) {
-            for (int i = 0; i < batchSize; i++) {
-                elementOffsets.add(0);
-            }
-            return RunLengthEncodedBlock.create(elementType, null, batchSize);
-        }
-
-        int[] offsets = new int[batchSize + 1];
-        for (int i = 1; i < offsets.length; i++) {
-            offsets[i] = offsets[i - 1] + elementOffsets.getInt(i - 1);
-        }
-        return ArrayBlock.fromElementBlock(batchSize, new boolean[batchSize], offsets, block);
+        Field elementField = field.getChildren().get(0).get();
```

Comment:
This is because an array always has a single inner element and a map always has two inner elements (key and value). We do `isPresent ` check for structs because some parameters can be missing due to schema evolution.